tests/unit/v1/test_charm_v1.py

```diff
@@ -0,0 +1,429 @@
+        state = State(
+            leader=True,
+            config={
+                "server": "https://acme-v02.api.letsencrypt.org/directory",
+            },
+        )
+        out = self.ctx.run("collect-unit-status", state)
+        assert out.unit_status == BlockedStatus("email address was not provided")
+
+    def test_given_server_not_provided_when_update_config_then_status_is_blocked(self):
+        state = State(
+            leader=True,
+            config={
+                "email": "banana@gmail.com",
+            },
+        )
+        out = self.ctx.run("collect-unit-status", state)
+        assert out.unit_status == BlockedStatus("ACME server was not provided")
+
+    def test_given_secret_id_not_provided_when_update_config_then_status_is_blocked(self):
+        state = State(
+            leader=True,
+            config={
+                "email": "banana@gmail.com",
+                "server": "https://acme-v02.api.letsencrypt.org/directory",
+            },
+        )
+        out = self.ctx.run("collect-unit-status", state)
+        assert out.unit_status == BlockedStatus("plugin configuration secret was not provided")
+
+    def test_given_invalid_email_when_update_config_then_status_is_blocked(self):
+        state = State(
+            leader=True,
+            secrets=[Secret(id="1", contents={0: {"api-key": "apikey123"}})],
+            config={
+                "email": "invalid email",
+                "server": "https://acme-v02.api.letsencrypt.org/directory",
+                "example-config-secret-id": "1",
+            },
+        )
+        out = self.ctx.run("collect-unit-status", state)
+        assert out.unit_status == BlockedStatus("invalid email address")
+
+    def test_given_invalid_server_when_update_config_then_status_is_blocked(self):
+        state = State(
+            leader=True,
+            secrets=[Secret(id="1", contents={0: {"api-key": "apikey123"}})],
+            config={
+                "email": "example@email.com",
+                "server": "Invalid ACME server",
+                "example-config-secret-id": "1",
+            },
+        )
+        out = self.ctx.run("collect-unit-status", state)
+        assert out.unit_status == BlockedStatus("invalid ACME server")
+
+    def test_given_invalid_plugin_config_when_update_status_then_status_is_blocked(self):
+        state = State(
+            leader=True,
+            secrets=[Secret(id="1", contents={0: {"api-key": "apikey123"}})],
+            config={
+                "email": "example@email.com",
+                "server": "https://acme-v02.api.letsencrypt.org/directory",
+                "example-config-secret-id": "1",
+            },
+            relations=[Relation(endpoint=CERTIFICATES_RELATION_NAME)],
+        )
+        with self.ctx.manager("collect-unit-status", state) as manager:
+            manager.charm.valid_config = False  # type: ignore
```

Comment:
Why take this approach instead of actually using juju configs and validating them? This would be cleaner and better integrated with scenario/Juju.

Reply:
The fake charm definition was copy pasted from v0, and this is the "canonical" way to modify the actual charm variables using scenario. I agree it should be that way. Changed.